app/src/test/java/com/eis/geoCalendar/database/AbstractEventDatabaseTest.java

```diff
@@ -0,0 +1,183 @@
+package com.eis.geoCalendar.database;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import org.robolectric.RuntimeEnvironment;
+
+import java.lang.reflect.Field;
+import java.util.List;
+import java.util.Map;
+
+import edu.emory.mathcs.backport.java.util.Arrays;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.junit.Assert.assertNotEquals;
+
+/**
+ * Class testing {@link AbstractEventDatabase}.
+ * Due to the fact the class is generalized to hold any kind of String (through
+ * {@link StringEntity}), nothing related to Events will be tested here.
+ *
+ * @author Riccardo De Zen
+ */
+@RunWith(RobolectricTestRunner.class)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Class 'AbstractEventDatabaseTest' must be declared as 'abstract'.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4822079)